internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go

```diff
@@ -259,6 +259,7 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
 		mockSpokeClient = spoke_k8s_client.NewMockSpokeK8sClient(mockCtrl)
 		mockSpokeClientCache = NewMockSpokeClientCache(mockCtrl)
 
+		ascProperties = make(map[string]interface{})
```

Comment:
If this is only relevant in an inner context then it should be defined and set in that context.

Moving it further away from where it's used makes the tests harder to understand.